quantumcat/circuit/circuit.py

```diff
@@ -110,89 +284,323 @@ def rz_gate(self, phi, qubit):
     #     self.operations.append({OpType.ecr_gate: [[qubit1], [qubit2]]})
 
     def s_gate(self, qubit):
+        """
+        Produces a single qubit rotation of pi/2 around the Z axis.
+        It is a Clifford gate and is known as root of Z gate.
+
+        Args:
+            qubit: index of qubit, to which the gate is to be applied.
+
+        Returns:
+            self: modified state of the qubit to the circuit.
+
+        Raises:
+            CircuitError: if qubit index is not within range.
+        """
         self.check_qubit_boundary(qubit)
         self.operations.append({OpType.s_gate: [qubit]})
         return self
 
     def sdg_gate(self, qubit):
+        """
+        Produces a single qubit rotation of -pi/2 around the Z axis.
+        It is a Clifford gate and is known as root of Z gate.
+
+        Args:
+            qubit: index of qubit, to which the gate is to be applied.
+
+        Returns:
+            self: modified state of the qubit to the circuit.
+
+        Raises:
+            CircuitError: if qubit index is not within range.
+        """
+
         self.check_qubit_boundary(qubit)
         self.operations.append({OpType.sdg_gate: [qubit]})
         return self
 
     def swap_gate(self, qubit1, qubit2):
+        """
+        Produces an interchange in the states of the target qubits.
+        It is symmetric and a Clifford gate.
+
+        Args:
+            qubit1: index of qubit, to which the gate is to be applied.
+            qubit2: index of qubit, to which the gate is to be applied.
+
+        Returns:
+            self: modified state of the qubit to the circuit.
+
+        Raises:
+            CircuitError: if qubit index is not within range.
+        """
         self.check_qubit_boundary(qubit1)
         self.check_qubit_boundary(qubit2)
         self.operations.append(({OpType.swap_gate: [[qubit1], [qubit2]]}))
         return self
 
-    def iswap_gate(self, qubit_a, qubit_b):
+    def iswap_gate(self, qubit1, qubit2):
+        """
+        Produces an interchange in the states of the target qubits and also adds a phase of i,
+        to the amplitudes of |01> and |10>.
+        It is symmetric and a Clifford gate.
+
+        Args:
+            qubit1: index of qubit, to which the gate is to be applied.
+            qubit2: index of qubit, to which the gate is to be applied.
+
+        Returns:
+            self: modified state of the qubit to the circuit.
+
+        Raises:
+            CircuitError: if qubit index is not within range.
+        """
         self.check_qubit_boundary(qubit_a)
         self.check_qubit_boundary(qubit_b)
         self.operations.append(({OpType.iswap_gate: [[qubit_a], [qubit_b]]}))
```

Comment:
will give error on run, since iswap() takes qubit1, qubit2 as parameter but qubit_a and qubit_b is used in the definition.